### PR TITLE
DOC-4345 added testable version of home page JSON example

### PR DESCRIPTION
**Pull request by andy-stark-redis**
This is part of an effort to get the [JSON examples on the "home" page](https://redis.io/docs/latest/develop/connect/clients/dotnet/#example-indexing-and-querying-json-documents) consistent for all the client libraries.

The current code in just added inline in the doc page, so it doesn't get the benefit of being tested in the NRedisStack repo and also doesn't have the usual tabbed code example features (like expanding to see the whole source file, etc). The code in this PR is basically the same as the existing code but with tests added.

### CODE REVIEW

File: tests/Doc/HomeJsonExample.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
+// EXAMPLE: cs_home_json
+
+// STEP_START import
+using NRedisStack.RedisStackCommands;
+using NRedisStack.Search;
+using NRedisStack.Search.Aggregation;
+using NRedisStack.Search.Literals.Enums;
+using NRedisStack.Tests;
```
Comment (atakavci) [inline tests/Doc/HomeJsonExample.cs:8]: would move this into 'HIDE' region.
at first sight i was thinking this should go into 'REMOVE' region, but then i realized
we left some test project related stuff in 'HIDE' regions overall the docs, is it on purpose ?

Comment (atakavci) [inline tests/Doc/HomeJsonExample.cs:8]: would move this into 'HIDE' region.
at first sight i was thinking this should go into 'REMOVE' region, but then i realized
we left some test project related stuff in 'HIDE' regions overall the docs, is it on purpose ?

Comment (andy-stark-redis) [reply tests/Doc/HomeJsonExample.cs:8]: > would move this into 'HIDE' region. at first sight i was thinking this should go into 'REMOVE' region, but then i realized we left some test project related stuff in 'HIDE' regions overall the docs, is it on purpose ?

@atakavci Yes, I've done this on purpose for this example. It's actually for this [introductory example](https://redis.io/docs/latest/develop/connect/clients/dotnet/#example-indexing-and-querying-json-documents) on the NRedisStack "home" page. The existing example is written to mention everything right from the start (including imports, etc). You are right, though, that in most other examples, we try to focus on just the relevant commands, not the boilerplate.

**Review by atakavci — APPROVED** [review]
looks good, thanks

**Pull request merged**